library/nmstate_l3_interface.py

```diff
@@ -120,14 +116,29 @@ def set_ipv4_addresses(interface_state, ipv4, purge=False):
     else:
         addresses = ipconfig.setdefault('addresses', [])
 
-    ip, prefix = ipv4.split('/')
-    addr = {'ip': ip, 'prefix-length': int(prefix)}
+    addr = create_ip_dict(ipv4)
     if addr not in addresses:
         addresses.append(addr)
 
     return interface_state
 
 
+def remove_ipv4_address(interface_state, ipv4):
+    ipconfig = interface_state.get('ipv4')
+    if not ipconfig:
+        return interface_state
+
+    addresses = ipconfig.get('addresses')
```

Comment:
It can be shorten and use  "try don't ask":
```
addresses = interface_state.get('ipv4', {}).get('addresses', [])
try:
    addresses.remove(create_ip_dict(ipv4))
except ValueError:
    pass
```

Reply:
I can live with EAFP here but I prefer the longer code because it is easier to comprehend and shows that there is nothing to do when there is no ipv4 state/no addresses at all.

Reply:
add made it EAFP.

Reply:
Ok, thanks. But I still would recommend getting rid of the noisy branches, it adds options which can be avoided.